### PR TITLE
Qt/Input: Improved how mouse movements are mapped to analog

### PR DESCRIPTION
**Pull request by sideprojectslab**
### Description of Changes
The pointer "scale" sliders have been replaced with pointer "speed" sliders. The hitgher the "speed" setting, the further will the stick move for the same mouse movement speed.

"Dead-Zone" sliders have been added in order to remove the inherent dead zone that a game imposes on an analog stick. When the mouse moves in any direction, the stick will immediately "jump over" the stated dead zone, so that even small mouse movements will cause valid (effective) stick positions.

A pointer "inertia" slider has been added. With higher inertia, the stick position will gradually return to zero (as opposed to immediately) IF AND ONLY IF the mouse has moved faster than the maximum available stick position range.

### Rationale behind Changes
This PR should make FPS games much more playable with mouse and keyboard. For small-medium mouse movements, aiming should feel almost identical to a true mouse-controlled PC game

### Suggested Testing Steps
Test with FPS games. Maximize in-game stick sensitivity and then adjust pointer speed in emulator settings until it "feels right". Dead-Zone should be increased from zero until the mouse feels responsive even for tiny and slow movements.

### CODE REVIEW

**Comment by sideprojectslab** [comment]
@refractionpcsx2 I couldn't for the life of me rebase the previous PR, so I made a new clean one, including your latest input about "delta"

**Comment by JordanTheToaster** [comment]
Unfortunately it still requires you to change all the values at least once for mouse movement to work properly.

**Comment by refractionpcsx2** [comment]
The UI changes seem to have put everything inside a box/group of some sort?  Causing everything to be inset, unlike before. Can you remove that? was there a reason for it?

**Comment by sideprojectslab** [comment]
the way I see it, all these settings should be inside a scroll area, so that one is free to resize the window freely while still having access to all functionality, think of it as a safety line. The scroll area is not visible by default, but only when the window is resized below the minimum necessary size. I found that this is important because the size of these windows depends on the display scaling factor set under windows. In my case, I use a fairly high scaling factor (150%) and with that, even just adding a tiny bit of vertical height to the window as a consequence of adding the new settings was completely breaking things (parts of the window outside the screen, weird glitches and so on)

**Comment by sideprojectslab** [comment]
The new settings are significantly different than the old ones and even the "speed" setting, that replaces the "sensitivity" (scale under the hood) carries a different meaning than in the past. So it is expected that things need to be manually adjusted. The only way to avoid that would be to include an automatic migration script, which seems quite complicated, but if anyone knows how to do it then feel free to ask and I'll let you contribute to the fork

File: pcsx2/Frontend/InputManager.cpp [hunk 1]
```diff
@@ -933,19 +936,36 @@ bool InputManager::PreprocessEvent(InputBindingKey key, float value, GenericInpu
 
 void InputManager::GenerateRelativeMouseEvents()
 {
+	static float pos[2] = {0.0f, 0.0f};
```
Comment (stenzek) [inline pcsx2/Frontend/InputManager.cpp:939]: Probably should put this at the top of the file (similar to the other variables), just for consistency.

Comment (sideprojectslab) [reply pcsx2/Frontend/InputManager.cpp]: ok, done that :)

**Comment by JordanTheToaster** [comment]
Mouse movement still does not work until you change the values at least once.

File: pcsx2/Frontend/InputManager.cpp [hunk 2]
```diff
@@ -1190,14 +1209,16 @@ void InputManager::ReloadBindings(SettingsInterface& si, SettingsInterface& bind
...
-		// From lilypad: 1 mouse pixel = 1/8th way down.
-		const float default_scale = (axis <= static_cast<u32>(InputPointerAxis::Y)) ? 8.0f : 1.0f;
-		s_pointer_axis_scale[axis] =
-			1.0f /
-			std::max(si.GetFloatValue("Pad", fmt::format("Pointer{}Scale", s_pointer_axis_names[axis]).c_str(), default_scale), 1.0f);
+		s_pointer_axis_speed[axis]     = si.GetFloatValue("Pad", fmt::format("Pointer{}Speed", s_pointer_axis_names[axis]).c_str()) / ui_ctrl_range * pointer_sensitivity;
+		s_pointer_axis_dead_zone[axis] = std::min(si.GetFloatValue("Pad", fmt::format("Pointer{}DeadZone", s_pointer_axis_names[axis]).c_str()) / ui_ctrl_range, 1.0f);
+		s_pointer_axis_range[axis]     = 1.0f - s_pointer_axis_dead_zone[axis];
 	}
+	s_pointer_inertia = si.GetFloatValue("Pad", fmt::format("PointerInertia").c_str()) / ui_ctrl_range;
```
Comment (stenzek) [inline pcsx2/Frontend/InputManager.cpp:1220]: This is missing a default value (and might be responsible for what Jordan said).

Comment (sideprojectslab) [reply pcsx2/Frontend/InputManager.cpp:1220]: I'll check that, thanks

Comment (sideprojectslab) [reply pcsx2/Frontend/InputManager.cpp:1220]: what I am actually seeing right now is that the emulator won't load any custom profile, but only the "shared" profile will work, even if I create the new profile from scratch. Can someone lend some help?

Comment (RedDevilus) [reply pcsx2/Frontend/InputManager.cpp:1220]: Did you change it in 2 locations? Globally:
image

Then per-game:
image

Unless that doesn't work with the USB devices.

Comment (refractionpcsx2) [reply pcsx2/Frontend/InputManager.cpp:1220]: > Unless that doesn't work with the USB devices.

#8051

Comment (RedDevilus) [reply pcsx2/Frontend/InputManager.cpp:1220]: Oh, nvm thought that was merged already.

Comment (stenzek) [reply pcsx2/Frontend/InputManager.cpp:1220]: > I'm not sure I follow, you mean a default value where it is declared as float at the top of the file?

No, the third parameter to `GetFloatValue()` is what is returned when the key does not exist in the config. (also, the fmt is redundant there)

Comment (sideprojectslab) [reply pcsx2/Frontend/InputManager.cpp:1220]: Ok thanks. Sorry for the extremely long reaction times, life has a way of getting in the way of what's truly important :)

**Comment by sideprojectslab** [comment]
Can you check if it works now? I made a clean build, mapped a few keys from scratch and got the mouse to work without having to move any slider. As for the custom profile what RedDevilus said fixed the issue

**Review by JordanTheToaster — APPROVED** [review]

**Comment by JordanTheToaster** [comment]
All works good now.

**Pull request merged**